src/main/java/org/jenkinsci/plugins/durabletask/ProcessLiveness.java

```diff
@@ -43,6 +54,23 @@
      * @return true if it is apparently still alive (or we cannot tell); false if it is believed to not be running
      */
     public static boolean isAlive(VirtualChannel channel, int pid, Launcher launcher) throws IOException, InterruptedException {
+        Boolean working = workingLaunchers.get(launcher);
+        if (working == null) {
+            // Check to see if our logic correctly reports that an unlikely PID is not running.
+            working = !_isAlive(channel, 9999, launcher);
```

Comment:
> to get the PID of the current process

Easily done via `getpgid(0)`.

> and you could use fork...join on a JVM that outputs its own PID to get a PID that does not exist

Oof, way too complicated.

Again, the impact of process 9999 existing is very minor. It simply means that some robustness code is disabled on this slave. If people see this unlikely event as a stumbling block, I would rather just disable this part of the check, and rely only on the check that `getpgid(0) != -1`.